src/app/models/field/__tests__/dateField.spec.ts

```diff
@@ -0,0 +1,98 @@
+import merge from 'lodash/merge'
+import { Model, Schema } from 'mongoose'
+import { DateFieldBase, FormResponseMode } from 'shared/types'
+
+import { IDateFieldSchema } from 'src/types'
+
+import dbHandler from 'tests/unit/backend/helpers/jest-db'
+
+import createDateFieldSchema from '../dateField'
+
+describe('models.fields.dateField', () => {
+  // Required as the field validator has a this.parent() check for response mode.
+  let MockParent: Model<{
+    responseMode: FormResponseMode
+    field: IDateFieldSchema
+  }>
+
+  beforeAll(async () => {
+    const db = await dbHandler.connect()
+    const dateFieldSchema = createDateFieldSchema()
+    MockParent = db.model(
+      'mockParent',
+      new Schema({
+        responseMode: {
+          type: String,
+          enum: Object.values(FormResponseMode),
+        },
+        field: dateFieldSchema,
+      }),
+    )
+  })
+  beforeEach(async () => await dbHandler.clearDatabase())
+  afterAll(async () => await dbHandler.closeDatabase())
+
+  it('should set default empty array for invalid days of the week when date field does not specify', async () => {
+    // Arrange
+    const mockDateField = {
+      dateValidation: {
+        selectedDateValidation: null,
+        customMaxDate: null,
+        customMinDate: null,
+      },
+    }
+    const expectedDateField: Partial<DateFieldBase> = {
+      dateValidation: {
+        selectedDateValidation: null,
+        customMaxDate: null,
+        customMinDate: null,
+      },
+      invalidDaysOfTheWeek: [],
+    }
+
+    // Act
+    const actual = await MockParent.create({
+      responseMode: FormResponseMode.Encrypt,
+      field: mockDateField,
+    })
+
+    // Assert
+    const expected = merge(expectedDateField, mockDateField, {
+      _id: expect.anything(),
+    })
+    expect(actual.field.toObject()).toEqual(expected)
+  })
+
+  it('should set expected array for invalid days of the week when date field specifies', async () => {
```

Comment:
should we add a negative test for rejection on invalid data? think back to equivalence partitioning in CS2103 😉

Reply:
Updated PR